pulumi/volumeReaper/src/cleanup.ts

```diff
@@ -0,0 +1,74 @@
+import k8s, { KubernetesObject } from '@kubernetes/client-node'
+import { VolumeSnapshot } from './snapshot'
+
+export const cleanup = async (
+  k8sApi: k8s.KubernetesObjectApi,
+  sts: string,
+  namespace: string,
+  pvcsToKeepCount: number
+) => {
+  const snapshots = await getExistingVolumeSnapshots(k8sApi, sts, namespace)
+  if (snapshots.length > pvcsToKeepCount) {
+    await removeOldSnapshots(k8sApi, snapshots, pvcsToKeepCount)
+  }
+
+  console.log('Snapshot cleanup completed')
+}
+
+const getExistingVolumeSnapshots = async (
+  k8sApi: k8s.KubernetesObjectApi,
+  sts: string,
+  namespace: string
+): Promise<KubernetesObject[]> => {
+  const snapshots = await k8sApi.list<VolumeSnapshot>(
+    'snapshot.storage.k8s.io/v1',
+    'VolumeSnapshot',
+    namespace,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    `statefulset=${sts}`
+  )
+  const items = snapshots.body.items
+  console.log(`Found ${items.length} snapshots for ${namespace}.${sts}`)
+  return items.sort(sortByCreationTimestampDesc)
+}
+
+const sortByCreationTimestampDesc = (a: VolumeSnapshot, b: VolumeSnapshot) =>
+  new Date(b.metadata?.creationTimestamp).getTime() - new Date(a.metadata?.creationTimestamp).getTime()
+
+const removeOldSnapshots = async (
+  k8sApi: k8s.KubernetesObjectApi,
+  snapshots: KubernetesObject[],
+  snapshotsToKeepCount: number
+) => {
+  await Promise.all(
+    getSnapshotsToRemove(snapshots, snapshotsToKeepCount).map(async (vc) => await deleteSnapshot(k8sApi, vc))
+  )
+}
+
+const deleteSnapshot = async (k8sApi: k8s.KubernetesObjectApi, snapshot: KubernetesObject) => {
+  await k8sApi.delete({
+    apiVersion: snapshot.apiVersion,
+    kind: snapshot.kind,
+    metadata: {
+      name: snapshot.metadata?.name,
+      labels: snapshot.metadata?.labels,
+    },
+  })
+  console.log(`Deleted ${snapshot.metadata?.name}`)
```

Comment:
```suggestion
      name: snapshot.metadata.name,
      labels: snapshot.metadata.labels,
    },
  })
  console.log(`Deleted ${snapshot.metadata.name}`)
```

Reply:
Could really just pass in `snapshot` thinking about it.
```ts
await k8sApi.delete(snapshot)
```

Reply:
Good catch, I've added the type all over the file